### PR TITLE
Allow BSC BNB and Polygon MATIC fiat on ramp

### PR DESCRIPTION
**Pull request by wachunei**
**Description**

This PR enables Buy button on Binance Smart Chain and Polygon with Transak as payment processor to get native asset (BNB, MATIC). 
Receive modal, Swaps alert and Send flow alert have been updated to show "Buy {{asset}}".

### CODE REVIEW

File: app/components/UI/FiatOrders/PaymentMethodSelector/index.ios.js [hunk 1]
```diff
@@ -103,9 +104,10 @@ function PaymentMethodSelectorView({
 					<SubHeader centered>{strings('fiat_on_ramp.purchase_method_title.wyre_sub_header')}</SubHeader>
 				)}
 			</Heading>
-
-			<WyreApplePayPaymentMethod onPress={onPressWyreApplePay} />
-			{network === '1' && <TransakPaymentMethod onPress={onPressTransak} />}
+			{(network === '1' || (network === '42' && Device.isIos())) && (
```
Comment (gantunesr) [inline app/components/UI/FiatOrders/PaymentMethodSelector/index.ios.js:107]: Can we make these chain ids constants? For example, `MAINNET_ID = 1`

File: app/components/UI/FiatOrders/index.js [hunk 2]
```diff
@@ -20,7 +21,8 @@ import processOrder from './orderProcessor';
 const POLLING_FREQUENCY = AppConstants.FIAT_ORDERS.POLLING_FREQUENCY;
 const NOTIFICATION_DURATION = 5000;
 
-export const allowedToBuy = (network) => network === '1' || (network === '42' && Device.isIos());
+export const allowedToBuy = (network) =>
+	network === '1' || (network === '42' && Device.isIos()) || TRANSAK_ALLOWED_NETWORKS.includes(network);
```
Comment (gantunesr) [inline app/components/UI/FiatOrders/index.js:25]: Same here

File: app/components/UI/FiatOrders/orderProcessor/transak.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ const TRANSAK_API_KEY_SECRET_PRODUCTION = process.env.TRANSAK_API_KEY_SECRET_PRO
  * @property {string} walletLink
  * @property {string} paymentOptionId Paymenth method ID, see: https://integrate.transak.com/Coverage-Payment-Methods-Fees-Limits-30c0954fbdf04beca68622d9734c59f9
  * @property {boolean} addressAdditionalData
- * @property {string} network this is NOT ethernet networks id
+ * @property {string} network this is NOT a chain id
```
Comment (gantunesr) [inline app/components/UI/FiatOrders/orderProcessor/transak.js:36]: Is this the network name? I understand that is not the chain id, but we can say specifically what it is

File: app/components/UI/FiatOrders/orderProcessor/transak.js [hunk 4]
```diff
@@ -217,18 +219,22 @@ export async function processTransakOrder(order) {
...
-			}),
-		[address]
-	);
+export const useTransakFlowURL = (address, network) => {
+	const params = useMemo(() => {
+		let [networks, cryptoCurrencyCode] = ['ethereum', 'ETH'];
+		if (network === '56') {
+			[networks, cryptoCurrencyCode] = ['bsc', 'BNB'];
+		} else if (network === '137') {
+			[networks, cryptoCurrencyCode] = ['polygon', 'MATIC'];
```
Comment (gantunesr) [inline app/components/UI/FiatOrders/orderProcessor/transak.js:228]: We should have constants for all of these values

**Review by gantunesr — COMMENTED** [review]
Left some small comments

**Comment by wachunei** [comment]
@gantunesr Great catches, I resolved them all in 65d55d20bc4921b28fd78935873792d7cb5eff63

**Review by gantunesr — APPROVED** [review]
LGTM

**Comment by wachunei** [comment]
Superseded by #3607

**Pull request closed without merging**